src/Integration/Binding/ProjectRuleSetWriter.cs

```diff
@@ -87,15 +87,21 @@ private bool TryUpdateExistingProjectRuleSet(string solutionRuleSetPath, string
         /// <summary>
         /// Remove all rule set inclusions which exist under the specified root directory.
         /// </summary>
-        internal /* testing purposes */  static void RemoveAllIncludesUnderRoot(RuleSet ruleSet, string rootDirectory)
+        internal /* testing purposes */  void RemoveAllIncludesUnderRoot(RuleSet ruleSet, string rootDirectory)
         {
             Debug.Assert(!string.IsNullOrWhiteSpace(rootDirectory));
-            
+
+            string ruleSetRoot = PathHelper.ForceDirectoryEnding(Path.GetDirectoryName(ruleSet.FilePath));
+
             // List<T> required as RuleSetIncludes will be modified
             List<RuleSetInclude> toRemove = ruleSet.RuleSetIncludes
                                             .Where(include =>
                                             {
-                                                string fullIncludePath = PathHelper.ResolveRelativePath(include.FilePath, rootDirectory);
+                                                string fullIncludePath = PathHelper.ResolveRelativePath(include.FilePath, ruleSetRoot);
+                                                if (!this.FileSystem.FileExists(fullIncludePath))
```

Comment:
Another important missing check. Need to check the path we computed above is valid. This will not be the case when the rule set is including a 'well known' rule set such as MinimumRecommendedRules.ruleset